src/config.py

```diff
@@ -34,29 +34,40 @@ def deployment_key(self):
     def runtime_key(self):
         return self.cdf_runtime_credentials
 
-    @root_validator(skip_on_failure=True)
-    def check_credentials(cls, values):
+    @staticmethod
+    def _verify_credentials(env, values):
         project = values["cdf_project"]
         kwargs = {
             "base_url": values["cdf_base_url"],
             "client_name": "function-action-validator",
             "disable_pypi_version_check": True,
         }
-        for env in ["deployment", "runtime"]:
-            with warnings.catch_warnings():
-                warnings.filterwarnings("ignore", category=UserWarning)
-                client = CogniteClient(api_key=values[f"cdf_{env}_credentials"], **kwargs)
-            if not client.login.status().logged_in:
-                raise ValueError(f"Can't login with {env} credentials")
-
-            inferred_project = client.login.status().project
-            if project is None:
-                logger.warning(f"Inferred project: {inferred_project} from given {env} credentials ")
-            if inferred_project != project:
-                raise ValueError(
-                    f"Inferred project, {inferred_project}, from the provided {env} credentials "
-                    f"does not match the given project: {project}"
-                )
+        with warnings.catch_warnings():
+            warnings.filterwarnings("ignore", category=UserWarning)
+            client = CogniteClient(api_key=values[f"cdf_{env}_credentials"], **kwargs)
+        if not client.login.status().logged_in:
+            raise ValueError(f"Can't login with {env} credentials")
+
+        inferred_project = client.login.status().project
+        if project is None:
+            logger.warning(f"Inferred project: {inferred_project} from given {env} credentials ")
+        elif inferred_project != project:
+            raise ValueError(
+                f"Inferred project, {inferred_project}, from the provided {env} credentials "
+                f"does not match the given project: {project}"
+            )
+        return inferred_project
+
+    @root_validator(skip_on_failure=True)
+    def check_credentials(cls, values):
+        deploy_key = cls._verify_credentials("deployment", values)
```

Comment:
Variable name `_key` is a bit misleading 😅

Reply:
Key as in API key - what would be better / should I just add in `_api_`?

Reply:
Doesn't `_verify_credentials` return inferred project?

Reply:
Lol now I get your point